lips/lip-9.md

```diff
@@ -0,0 +1,95 @@
+---
```

Comment:
General comments:
1. It's unclear to me why this is desirable to have on chain.  If my key were leaked, now all of my purchase information is parsable.
2. If we wanted to have this rich metadata, why wouldn't we prefer to do it over off-chain? It keeps it private and allows for richer data than what we would want to ever store on chain (for example, maybe an image)
3. I'm uncertain if users will ever really want to proactively share their purchase information with a 3rd-party.  I wonder if we should wait until we see a real use case for this?
4. One of the things you mention is portability of transaction information.  I agree that there is value there.  But not sold on this being the way to do it.  This feels more dangerous.  Why not use off-chain for dumping data and importing data?  Francois has been thinking about this as well, let me ping him to chime in here
5. For resolving disputes, wouldn't the VASP generally be able to do this?  So if you exchanged this data via off-chain, the VASPs would have the data already and could do dispute resolution
6. If we did want this to be on-chain, why include the raw data instead of just signature of the hash of the data?  That way, others couldn't reconstruct it
7. I worry that this could easily cause confusion.  Some company says "share your private key with us for $5 so we can analyze your payments" and now I accidentally share my wallet private key.  I worry that the potential risks are higher than any benefits from it

Reply:
Thanks for your feedback.
Generally speaking, the concept of storing such data on-chain and not off-chain is for giving the users more power and control over their data. When storing it off-chain, the user actually relies on the wallet/merchant VASP(s) to give the data when they asked to do it. This data belongs to the user, not the VASP. 
Under this assumption, let me address your points:  

1. That's right. I believe that most wallets and vendors will store your private key for you and let you identify yourself using conventional authentication methods. But when you will want to drop from your current wallet vendor and move to another one, it should be a fairly straightforward process to export your personal PK and import it to the new vendor.
2. Fair point. I want it to be on-chain for the above reasons. Going off-chain will let us store richer data.
3. There are already a large collection of 3rd party apps that scans your bank account and cc statements to give you insights and suggestions for your financial behavior. I think it would be very useful for such use cases. 
4. Making it off-chain is possible. in such a case we will have to define a standard way of export/import. The main issue will be how we can enforce these rules on the VASPs. 
5. Sure. Although the only thing that is signed is the reference_id of the off-chain protocol. If we want to move it to off-chain I think we should add one or more steps to wrap the whole payment with undisputable data from both sides. 
6. Including only hashes would still leave the actual data (and power) in the VASP(s) hands. I don't see any difference between such a hash and the currently suggested metadata (reference_id)
7. If you're thinking of unhosted wallets - that's correct. But a similar concern should be raised for the compliance key as well. If we're talking about custodial wallets, I don't think we should worry about it as the wallet private key will not be exposed to the end-user anyway. However, I did use different encryption algorithms with different key schemes to make such confusion harder.